Docs: Configure security hardening (#58738)
grafana/grafana: docs/sources/setup-grafana/configure-security/configure-security-hardening.md

---
aliases:
  - /docs/grafana/latest/setup-grafana/configure-security/configure-security-hardening/
description: Security hardening enables you to apply additional security which might stop certain vulnerabilities from being exploited by a malicious attacker. 
title: Configure security hardening
---

# Configure  security hardening

Security hardening enables you to apply additional security which might stop certain vulnerabilities from being exploited by a malicious attacker. 

> > **Note:** These settings will be found in the [grafana.ini configuration file]({{< relref "../configure-grafana/#configuration-file-location" >}}). After the file has been edited, the Grafana server needs to be restarted in order to apply the changes.

Reviewer: ```suggestion
> **Note:** These settings are available in the [grafana.ini configuration file]({{< relref "../configure-grafana/#configuration-file-location" >}}). To apply changes to the configuration file, restart the Grafana server.
```

---
aliases:
    - /docs/grafana/latest/setup-grafana/configure-security/configure-security-hardening/
description: Security hardening enables you to apply additional security which might stop certain vulnerabilities from being exploited by a malicious attacker.
title: Configure security hardening
---

# Configure security hardening

Security hardening enables you to apply additional security which might stop certain vulnerabilities from being exploited by a malicious attacker.

## Cookies

If Grafana uses HTTPS, it is possible to further secure the cookie that is used as authentication to access the Web UI.

> **Note:** Grafana must use HTTPS in order for the below configurations to work properly.

### Secure cookie attribute

Set the _cookie_secure_ option to `true`. This will enforce clients to set and send the cookie over a (valid HTTPS) secure connection to help mitigate against some MITM-attacks.

Example:

> ```toml

Reviewer: which toml file?

Author: I added a note in the beginning regarding which file to be modified.

> # set to true if you host Grafana behind HTTPS. default is false.

Reviewer: ```suggestion
# Set to true if you host Grafana behind HTTPS. The default value is false.
```

Example:

```toml
# Login cookie name
login_cookie_name = __Host-grafana_session
```

## Security headers

Grafana has a few additional headers that can be configured which might help to mitigate against certain attacks such as XSS.

### Content Security Policy

The default CSP template is already configured to provide sufficient protection against some attacks.

Example:

```toml
# Enable adding the Content-Security-Policy header to your requests.
# CSP allows to control resources the user agent is allowed to load and helps prevent XSS attacks.
content_security_policy = true

# Set Content Security Policy template used when adding the Content-Security-Policy header to your requests.
# $NONCE in the template includes a random nonce.
# $ROOT_PATH is server.root_url without the protocol.
content_security_policy_template = """script-src 'self' 'unsafe-eval' 'unsafe-inline' 'strict-dynamic' $NONCE;object-src 'none';font-src 'self';style-src 'self' 'unsafe-inline' blob:;img-src * data:;base-uri 'self';connect-src 'self' grafana.com ws://$ROOT_PATH wss://$ROOT_PATH;manifest-src 'self';media-src 'none';form-action 'self';"""
```

## Additional security hardening

> ### Hide version

Reviewer 2: I think the titles should be swapped: L76 with L88.

Author: Thanks, good catch, fixed in https://github.com/grafana/grafana/pull/58738/commits/c401ffc47c9961af4291d05807759b6aef047786
